spec/acceptance/snapshots_spec.rb

```diff
@@ -31,6 +31,10 @@
     status('Test: restore snapshot')
     expect(environment.execute('vagrant', 'snapshot', 'restore', 'test')).to exit_with(0)
 
+    # KVM needs a moment for IO to catch up.
+    # If anyone comes up with a better way to wait, please update this.
+    sleep(3)
```

Comment:
I suspect an email to get libvirt devs would be needed. I assumed that revert was blocking, but perhaps that doesn't cover flushing or the state being fully switched. Hopefully they could enlighten on what would need to be done after the revert call to confirm the revert is fully completed.

This should be good enough for the tests for now though.